client/dom/genesearch.ts

```diff
@@ -133,6 +134,11 @@ export function addGeneSearchbox(arg: GeneSearchBoxArg) {
 	const tip = arg.tip,
 		row = arg.row
 
+	if (arg.geneOnly && arg.snpOnly) {
+		row.append('span').text('.geneOnly and snpOnly cannot both be true')
```

Comment:
Not a good use of boolean flags, should be converted to maybe `arg.restriction: 'gene' | 'snp' | 'none'``. The boolean flag requires validation whereas using the same arg attribute would not require validation. The boolean flag also will require a more  complicated or less intuitive type definition.

Reply:
yes will do this refactor later